CONTRIBUTING.md

```diff
@@ -1,31 +1,117 @@
-3. Write and test your code.
-4. Ensure that your code compiles with the provided build engine and update the provided build engine as part of your PR where needed and where appropriate.
-5. Write test cases for your code and include those in `src/test` such that they are runnable from the command line using `./rippled -u`. (Some changes will not be able to be tested this way.)
-6. Ensure your code passes automated checks (e.g. clang-format and levelization.)
-7. Squash your commits (i.e. rebase) into as few commits as is reasonable to describe your changes at a high level (typically a single commit for a small change.)
-8. Open a PR to the main repository onto the _develop_ branch, and follow the provided template.
-
-# Major Changes
-If your code change is a major feature, a breaking change or in some other way makes a significant alteration to the way the XRPL will operate, then you must first write an XLS document (XRP Ledger Standard) describing your change.
-To do this:
-1. Go to [XLS Standards](https://github.com/XRPLF/XRPL-Standards/discussions).
-2. Choose the next available standard number.
-3. Open a discussion with the appropriate title to propose your draft standard.
-4. Link your XLS in your PR.
+
+We assume you are familiar with the general practice of [making contributions
+on GitHub][1].
+This file includes only special instructions specific to this project.
+
+
+## Before you start
+
+All of your contributions must be developed in your personal
+[fork](https://github.com/XRPLF/rippled/fork).
+No personal branches may ever be pushed to the [main project][rippled].
+These are the only branches that may ever exist in the main project:
+
+- `develop`: The latest set of unreleased features, and the most common
+    starting point for contributions.
+- `release`: The latest release candidate.
+- `master`: The latest stable release.
+- `gh-pages`: The documentation for this project, built by Doxygen.
+
+All of your commits must be signed,
+and GitHub must know your verifying key.
+Please walk through the excellent documentation from GitHub to set
+up [signature verification][signing].
+
+[rippled]: https://github.com/XRPLF/rippled
+[signing]: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification
+
+
+## Major contributions
+
+If your contribution is a major feature or breaking change,
+then you must first write an XRP Ledger Standard (XLS) describing it.
+Go to [XRPL-Standards](https://github.com/XRPLF/XRPL-Standards/discussions),
+choose the next available standard number, and
+open a discussion with an appropriate title to propose your draft standard.
+
+When you submit a PR for this change, please link the corresponding XLS in the
+description.
```

Comment:
```suggestion
When you submit a PR for this change, please link the corresponding XLS draft in the description. As XLS drafts are considered work in progress and open for discussion, PRs should not be submitted without due time for discussions, questions, suggestions and additions to the XLS draft. It is the responsibility of the XLS author to update the draft to match the final implementation when the PR is merged.
```

Reply:
@thejohnfreeman this is precisely @Silkjaer's suggestion, just in the form of a GitHub "Suggested change" in case that is easier to see/adopt.